File: SalonDeEventos_FrontEnd/SalonDeEventos/src/app/modelo/imgSalon.ts
import { Salon } from "./salon";

export class ImgSalon {
    imgSalId: number|undefined;
    imgSalNombre: string;
    imgSalUrl: string;
    salId: Salon;
    
    constructor(
        imgSalId?: number,
        imgSalNombre?: string,
        imgSalUrl?: string,
        salId ?: Salon

    ) {
        this.imgSalId = imgSalId;
        this.imgSalNombre = imgSalNombre || "";
        this.imgSalUrl = imgSalUrl || "";
        this.salId = salId || new Salon;
    }

}

File: SalonDeEventosBack_End/salon/src/main/java/com/m5a/salon/model/entity/ImgSalon.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.m5a.salon.model.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import java.io.Serializable;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 *
 * @author LaptopSA
 */
@Entity
@Data
@AllArgsConstructor
@Table(name = "ImagenesSalones")
@NoArgsConstructor
public class ImgSalon implements Serializable {

    /**
     *
     */
    private static final long serialVersionUID = 1L;
    /**
     *
     */
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "imgSalId")
    private Long imgSalId;

    @Column(name = "imgSalNombre")
    private String imgSalNombre;

    @Column(name = "imgSalUrl")
    private String imgSalUrl;

    @ManyToOne
    @JoinColumn(name = "salId", referencedColumnName = "salId")
    private Salon salId;
}
